### DISCUSSION TITLE
请教下这样将多个MMKV文件迁移至一个，是否可以

### DISCUSSION THREAD

**Discussion by zlcm** [Tencent/MMKV#1174] [Q&A]
你好，请教下
iOS平台
项目前期对MMKV使用不规范，很多模块都自己生成了一个mmapID，再有些使用user+nmapID，导致项目中MMKV实例多达几十个。现在想规范下，迁移到几个MMKV文件中。MMKV没有提供相关的方法。所以我写了个迁移方法。
```objective-c
+ (void)move:(MMKV *)srcMMKV to:(MMKV *)dstMMKV {
    if (srcMMKV.allKeys.count == 0) return;
    [srcMMKV enumerateKeys:^(NSString * _Nonnull key, BOOL * _Nonnull stop) {
        auto data = srcMMKV->m_mmkv->getDataForKey(key);
        dstMMKV->m_mmkv->setDataForKey(std::move(data), key);
    }];
    [srcMMKV clearAll];
    [srcMMKV close];
}
```
这方法会报getDataForKey等方法为私有，所以我将其改为了public。
自己大概测了下，数据好像是可以成功迁移的。

想请教下，这样是否可以，有没有什么风险。

谢谢。

**Comment by lingol** [comment]
As long as your value is stored as Data inside, it's OK. NSString, NSData, & NSCoding objects are stored as Data. Others such as primary types (int, float, etc) are not.

**Reply by zlcm** [reply]
int  float 会有问题吗？
我都是以MMBuffer取出来，再存的。
int, float看源码最后也是以MMBuffer存的吧？
用MMKVDemo测试了bool, int32, uint32, int64, uint64, float, double，迁移到新的MMKV里后，也能正确的取到值，与原始值一至。
@lingol
